### PR TITLE
fix(connector): add config cleanup on payment connector deletion

### LINKED ISSUE #5966: [BUG] Bug in merchant connector account create - delete flows
### Bug Description

There is a problem with how the `configs` table is updated. The problem arises when a user deletes a connector and then adds the same one, the config isn't cleaned from the connector when it's deleted, and the new connector (with the same type) is appended to the array.
It looks something like this:
```json
[{"connector":"adyen","merchant_connector_id":"mca_123XXX"}, //old one, already deleted
{"connector":"adyen","merchant_connector_id":"mca_456XXX"}] //new one
``` 
When a payment (or payout or anything else connector related) is being created the router looks at the first occurrence of the connector (which is already deleted) and then looks at the `merchant_connector_account` table for an already deleted `merchant_connector_id`, unsurprisingly does not find it and throws an error. Affects both payment and payout processors

### Expected Behavior

I think either the payment connector delete function should clean the config array of the deleted connector, or the update function should prepend the config instead of appending. 

### Actual Behavior

The delete function does not clean the config and then just appends it when new creator of same type is added

### Steps To Reproduce

To reproduce: 
1. Create a merchant connector account of any type
2. Delete it
3. Create the same connector 
4. Try to make a payment with that connector
5. Get error 404:
```json
{
    "error": {
        "type": "invalid_request",
        "message": "Merchant connector account does not exist in our records",
        "code": "HE_02",
        "reason": "mca_123XXX does not exist"
    }
}
```

### Context For The Bug

Trace for update config:

```sh
at crates/diesel_models/src/query/generics.rs:247
    in router::db::configs::update_config_in_database
    in storage_impl::redis::cache::publish_and_redact
    in router::db::configs::update_config_by_key
    in core::routing::helpers::update_merchant_default_config
    in router::core::retrieve_and_update_default_fallback_routing_algorithm_if_routable_connector_exists
```

### Environment

Are you using hyperswitch hosted version? No

If not (or if building/running locally), please provide the following details:
1. Operating System or Linux distribution: Ubuntu 22.04
2. Rust version (output of `rustc --version`): `rustc 1.80.1 (3f5fd8dd4 2024-08-06)`
3. App version (output of `cargo r --features vergen -- --version`): `2024.08.27` (not running the latest version but checked the functions responsible and there weren't any changes)

### Have you spent some time checking if this bug has been raised before?

- [x] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

Yes, I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by noagbmn**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Fixes #5966 (in both v1 and v2). Now the router cleans up the configs table (or business_profile in v2) when a connector is deleted.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Added a payment and payout processor, checked if values are present in config, deleted and rechecked.
v1:
v1test

v2:
v2test

## Test on Integ
- Create any 2 MCA's
```
curl --location 'http://localhost:8080/account/merchant_1733401776/connectors' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data '{
    "connector_type": "payment_processor",
    "connector_name": "stripe",
    "connector_account_details": {
        "auth_type": "HeaderKey",
        "api_key": "{{api_key}}"
    },
     "connector_label": "stripe_US_default7777",
    "test_mode": false,
    "disabled": false,
    "payment_methods_enabled": [
        {
            "payment_method": "card",
            "payment_method_types": [
                {
                    "payment_method_type": "credit",
                    "card_networks": [
                        "Visa"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 200000,
                    "recurring_enabled": false,
                    "installment_payment_enabled": false
                },
                {
                    "payment_method_type": "debit",
                    "card_networks": [
                        "JCB"
                    ],
                    "minimum_amount": 1,
                    "maximum_amount": 200000,
                    "recurring_enabled": false,
                    "installment_payment_enabled": false
                }
            ]
        },
        {
            "payment_method": "wallet",
            "payment_method_types": [
                {
                    "payment_method_type": "google_pay",
                    "payment_experience": "invoke_sdk_client",
                    "card_networks": null,
                    "minimum_amount": 1,
                    "maximum_amount": 50000,
                    "recurring_enabled": false,
                    "installment_payment_enabled": false
                },
                {
                    "payment_method_type": "apple_pay",
                    "payment_experience": "invoke_sdk_client",
                    "card_networks": null,
                    "minimum_amount": 600,
                    "maximum_amount": 800,
                    "recurring_enabled": false,
                    "installment_payment_enabled": false
                }
            ]
        },
        {
            "payment_method": "pay_later",
            "payment_method_types": [
                {
                    "payment_method_type": "affirm",
                    "payment_experience": "redirect_to_url",
                    "card_networks": null,
                    "minimum_amount": 1,
                    "maximum_amount": 50000,
                    "recurring_enabled": false,
                    "installment_payment_enabled": false
                }
            ]
        }
    ],
    "metadata": {
        "city": "NY",
        "unit": "245"
    }
    
        
}'
```
- Do the list for default fallback routing connectors, the connector should be present
```
curl --location 'http://localhost:8080/routing/default/profile' \
--header 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYWQyMTRiZjUtMjgyYS00OWRiLWIzMjEtZGM3ZTExODRlNTliIiwibWVyY2hhbnRfaWQiOiJzZW5kZXJfbmV0Iiwicm9sZV9pZCI6ImludGVybmFsX3ZpZXdfb25seSIsImV4cCI6MTcyMTMxMDI0NCwib3JnX2lkIjoib3JnX2JmQXZqZWZRMG5oTjVJVm95VTZGIn0.hCH_Pd0d2d08ESOUy1PMGfyuh5TzXQLfi0wBIev1O6U' \
--data ''
```
- Delete the one of those 2 MCA's
```
curl --location --request DELETE 'http://localhost:8080/account/merchant_1733401776/connectors/mca_jIEg3gZxzs54aMMrhT9D' \
--header 'Accept: application/json' \
--header 'api-key: test_admin'
```
- The list for default fallback routing connectors, the connector shouldn't be present
```
curl --location 'http://localhost:8080/routing/default/profile' \
--header 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiYWQyMTRiZjUtMjgyYS00OWRiLWIzMjEtZGM3ZTExODRlNTliIiwibWVyY2hhbnRfaWQiOiJzZW5kZXJfbmV0Iiwicm9sZV9pZCI6ImludGVybmFsX3ZpZXdfb25seSIsImV4cCI6MTcyMTMxMDI0NCwib3JnX2lkIjoib3JnX2JmQXZqZWZRMG5oTjVJVm95VTZGIn0.hCH_Pd0d2d08ESOUy1PMGfyuh5TzXQLfi0wBIev1O6U' \
--data ''
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by noagbmn** [comment]
@SanchithHegde Hi, is this PR going to get reviewed? The issue also didn't get labeled

**Review by Aprabhat19 — CHANGES_REQUESTED** [review]

File: crates/router/src/core/admin.rs [hunk 1]
```diff
@@ -1910,6 +1910,65 @@ impl<'a> MerchantDefaultConfigUpdate<'a> {
         }
         Ok(())
     }
+
+    async fn retrieve_and_delete_from_default_fallback_routing_algorithm_if_routable_connector_exists(
+        &self,
+    ) -> RouterResult<()> {
+        let mut default_routing_config = routing::helpers::get_merchant_default_config(
```
[the remark this answers is not in the document]
Comment (noagbmn) [reply crates/router/src/core/admin.rs:1917]: I'm not sure what you mean? Both `routing_default_merchat_id` and `routing_default_po_merchat_id` are still updated at https://github.com/juspay/hyperswitch/blob/05080259132fb12cdef40a999bd02b6fe2beeeaa/crates/router/src/core/admin.rs#L1866-L1875
so why shouldn't they be cleared when the connector is deleted?

[the remark this answers is not in the document]
Comment (Aprabhat19) [reply crates/router/src/core/admin.rs:1917]: Yeah we can keep it then

[the remark this answers is not in the document]
Comment (noagbmn) [reply crates/router/src/core/admin.rs:1917]: Should I re-request your review if we're keeping it? @Aprabhat19

[the remark this answers is not in the document]
Comment (Aprabhat19) [reply crates/router/src/core/admin.rs:1917]: Yes sure, will re-review it @noagbmn

**Comment by noagbmn** [comment]
@Aprabhat19 Hi again! Re-requested your review

**Review by Aprabhat19 — DISMISSED** [review]

**Comment by noagbmn** [comment]
@SanchithHegde Hi sorry can we get a second reviewer?

**Review by SanchithHegde — DISMISSED** [review]
Looks good to me!

Thanks for the contribution, and apologies for the delay!

File: crates/router/src/core/admin.rs [hunk 2]
```diff
@@ -3199,6 +3312,38 @@ pub async fn delete_connector(
             id: id.clone().get_string_repr().to_string(),
         })?;
 
+    let business_profile = crate::core::utils::validate_and_get_business_profile(
```
Comment (Narayanbhat166) [inline crates/router/src/core/admin.rs:3315]: you can get the business profile directly from the `profile_id` instead of using this function.

Comment (Narayanbhat166) [reply crates/router/src/core/admin.rs:3315]: use this function https://github.com/juspay/hyperswitch/blob/2a790cea3c792586ccba11f3e11fdd9b7a4e02e9/crates/router/src/db/business_profile.rs#L31

Comment (noagbmn) [reply crates/router/src/core/admin.rs:3315]: How though? For `v2` feature flag I don't see a Profile field or any implementation that return Profile

https://github.com/juspay/hyperswitch/blob/64383915bda5693df1cecf6cc5683e8b9aaef99b/crates/hyperswitch_domain_models/src/merchant_account.rs#L168-L181

Comment (noagbmn) [reply crates/router/src/core/admin.rs]: should be done

**Review by SanchithHegde — APPROVED** [review]

**Review by Aprabhat19 — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**